### ISSUE TITLE
📎 Implement formatting option to include space inside parens, braces, brackets, etc.

### ISSUE THREAD

**Issue by sirreal** [biomejs/biome#4607]
### Description

Add formatting option to include spacing inside parentheses, square brackets, curly braces, and angle brackets.

For example, this input:

```ts
function id<TypeArg>(x: TypeArg): TypeArg {
	return x;
}
const arr = [1, 2, 3];
if (id(true)) {
	// …
}
```

Would be formatted as follows with a "space inside" option enabled:

```ts
function id< TypeArg > ( x: TypeArg ): TypeArg {
	return x;
}
const arr = [ 1, 2, 3 ];
if ( id( true ) ) {
	// …
}
```

See https://github.com/biomejs/biome/discussions/2360#discussioncomment-10690814.

**Comment by DmitryFX** [comment]
Agree! Current formatting 'standard' lacks spacing so much (who decided this to be a standard, btw?).

**Comment by siketyan** [comment]
See our [Formatter Option Philosophy](https://biomejs.dev/formatter/option-philosophy/) and please open a discussion first.

**Comment by ematipico** [comment]
Why was this closed? We actually want to implement this https://github.com/biomejs/biome/discussions/2360#discussioncomment-10690814

**Comment by siketyan** [comment]
Sorry I missed that
